csharp/test/Microsoft.ML.OnnxRuntime.Tests.NetCoreApp/InferenceTest.netcore.cs

```diff
@@ -95,6 +95,7 @@ private void TestCUDAProviderOptions()
                 providerOptionsDict["do_copy_in_default_stream"] = "1";
                 providerOptionsDict["cudnn_conv_use_max_workspace"] = "1";
                 providerOptionsDict["cudnn_conv1d_pad_to_nc1d"] = "1";
+                providerOptionsDict["gelu_disable_half2"] = "1";
```

Comment:
Why do we need expose this to user?

Reply:
I mentioned a bit in the PR description, pasted here also :
"Expose gelu_disable_half2  in OrtCUDAProviderOptionsV2 from include/onnxruntime/core/providers/cuda/cuda_provider_options.h. because the CUDA kernel implementation need such a setting to choose the corresponding branch. Originally the flag comes from a static singleton of TransformerOptions in onnxruntime/contrib_ops/cuda/bert/transformer_common.h, it is specifically for the contrib_ops/cuda/bert folder, so letting onnxruntime/core/providers/cuda/tensor/gelu.cc reference such a header might not be a good option. (Let me know if you have different ideas)." 

Before having this option, here is my thinking of reusing original TransformerOptions.DisableHalf2() . TransformerOptions is set with env var now, looks only for internal test usage, it is in contrib_ops folder. I feel if we use it in core/providers/cuda, maybe we should more that definition in core instead contrib_ops/bert, while from its name it is only for transformers, so I am hesitating how to reuse it in a reasonable way. Would appreciate any suggestions .

Reply:
That option is used in testing only. It will tell how much benefit is brought by half2. You can remove the flag if needed.
It is not intended to be used by end user since enabling the flag will get better performance.

Reply:
Let me enable the half2 by defaul in the ONNX domain Gelu.cc, and leave the original FastGelu use original flag to enable or not. Thanks @tianleiwu.